src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -800,6 +821,23 @@ LIMIT 1
         return new GatewayModel.FungibleResourcesCollection(totalCount, previousCursor, nextCursor, items.Take(limit).ToList());
     }
 
+    private async Task<List<ResourceEntitySupplyHistory>> GetResourcesSupplyData(long[] entityIds, GatewayModel.LedgerState ledgerState, CancellationToken token)
+    {
+        var result = await _dbContext.ResourceEntitySupplyHistory.FromSqlInterpolated($@"
+WITH variables (entity_id) AS (SELECT UNNEST({entityIds}))
+SELECT resh.*
+FROM variables
+INNER JOIN LATERAL(
+    SELECT *
+    FROM resource_entity_supply_history
+    WHERE from_state_version <= {ledgerState.StateVersion} AND resource_entity_id = variables.entity_id
+    ORDER BY from_state_version DESC
+    LIMIT 1
+) resh ON true").AsNoTracking().ToListAsync(token);
+
+        return result;
```

Comment:
Isn't `_dbContext` a `ReadOnlyDbContext`? It's `.AsNoTracking()` by default so it's not needed to specify that explicitly.